### PR TITLE
[8.0][FIX]sale_commission: Only create commissions in customer invoices.

### PR DESCRIPTION
**Pull request by jesusVMayor**
Fixes the creation of account.invoice.line.agent in supplier invoices. It occurs when a customer with agent is also an agent.

### CODE REVIEW

File: sale_commission/models/account_invoice.py [hunk 1]
```diff
@@ -62,26 +62,26 @@ class AccountInvoiceLine(models.Model):
     _inherit = "account.invoice.line"
 
     @api.model
-    def _default_agents(self):
+    def create(self, vals):
```
Comment (pedrobaeza) [inline sale_commission/models/account_invoice.py]: You must not replace the default. It's there for something: modify in screen the default commission values if needed.

File: sale_commission_product/models/account_invoice.py [hunk 2]
```diff
@@ -17,21 +17,22 @@ def product_id_change(
...
-                commission_id_product = self.env["product.product.agent"]\
-                    .get_commission_id_product(product, agent)
-                if commission_id_product:
-                    commission_id = commission_id_product
-                agent_list.append({'agent': agent.id,
-                                   'commission': commission_id
-                                   })
-
-                res['value']['agents'] = [(0, 0, x) for x in agent_list]
+        if type == 'out_invoice':
```
Comment (ivantodorovich) [inline sale_commission_product/models/account_invoice.py]: what about out_refund?

**Comment by jesusVMayor** [comment]
Added out_refund.
I added a check in the settlement wizard because the only thing that prevents commissions creation for the supplier invoices its the partner_id in contexts it is not passed, and this could be changed by any module. I not see another way to improve the control of commissions creation without removing the default.

File: sale_commission_product/models/account_invoice.py [hunk 3]
```diff
@@ -17,21 +17,22 @@ def product_id_change(
...
-                commission_id_product = self.env["product.product.agent"]\
-                    .get_commission_id_product(product, agent)
-                if commission_id_product:
-                    commission_id = commission_id_product
-                agent_list.append({'agent': agent.id,
-                                   'commission': commission_id
-                                   })
-
-                res['value']['agents'] = [(0, 0, x) for x in agent_list]
+        if type in ('out_invoice', 'out_refund'):
```
Comment (pedrobaeza) [inline sale_commission_product/models/account_invoice.py]: Don't make 2 nested if, but put all conditions in one.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by atchuthan — APPROVED** [review]
minor change as per @pedrobaeza suggestion

Could change label to Needs fixing!!

**Comment by jesusVMayor** [comment]
Fixed

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**